vulcan/lib/server/models/figure.rb

```diff
@@ -76,5 +82,13 @@ def to_hash(storage: nil)
         updated_at: updated_at.iso8601,
       }
     end
+
+    private
+
+    def self.validate(payload, user, project_name)
+      project_permissions = user.permissions[project_name]
+
+      raise ArgumentError.new('Guests cannot save public figures.') if project_permissions[:role] == :guest && payload[:tags]&.include?('public')
```

Comment:
I wonder if guest users would understand what to do when they see this.  Should we explain/recommend copying the figure and removing the public tag in order to successfully save?  (And prioritize soon making a `save as copy` route)